syntax = "proto3";

package example;

import "google/api/annotations.proto";
import "google/protobuf/descriptor.proto";


// 定义你的包名

service ExampleService {
  rpc GetExample (ExampleRequest) returns (ExampleResponse) {
    option (google.api.http) = {
      get: "/v1/example/{id}"
    };
  }

  rpc Send (ETTRequest) returns (StandardResponse) {
    option (google.api.http) = {
      post: "/send"
    };
  }
  
  rpc Test(ETTRequest) returns (StandardResponse) {
    option (google.api.http) = {
      post: "/test"
    };
  }

  rpc Stop(NullRequest) returns (ExampleResponse) {
    option (google.api.http) = {
      post: "/stop"
    };
  }

}

message ExampleRequest {
  string id = 1;
}

message ExampleResponse {
  string message = 1;
}

message NullRequest {}

message ETTRequest {
  string date = 1;
  float HUFL = 2;
  float HULL = 3;
  float MUFL = 4;
  float MULL = 5;
  float LUFL = 6;
  float LULL = 7;
  float OT = 8;
}

message StandardResponse {
  string result = 1;
  string msg = 2;
}